File: Benchmarks/AlternatorComparison.cs
using System;
using System.Text.Patterns;
using System.Text.RegularExpressions;
using BenchmarkDotNet.Attributes;

namespace Benchmarks {
	[ClrJob, CoreJob, CoreRtJob]
	[MemoryDiagnoser]
	public class AlternatorComparison {

		readonly Pattern Alternator = (Pattern)"Hello" | "Goodbye";

		readonly Regex Regex = new Regex("^(?:Hello|Goodbye)");

		[Params("Hello", "Goodbye", "Failure")]
		public String Source { get; set; }

		[Benchmark(Baseline = true)]
		public Result Stringier() => Alternator.Consume(Source);

		[Benchmark]
		public Match MSRegex() => Regex.Match(Source);
	}
}

File: Benchmarks.FSharp/AlternatorComparison.fs
namespace Benchmarks

open System
open System.Text.Patterns
open FParsec
open BenchmarkDotNet.Attributes
open BenchmarkDotNet.Diagnosers

[<ClrJob>]
[<MemoryDiagnoser>]
type AlternatorComparison() =
    let pattern = "Hello" || "Goodbye"
    let parsec = attempt (pstringCI "Hello" <|> pstringCI "Goodbye")

    [<Params("Hello", "Goodbye", "Bacon")>]
    member val Source = "" with get, set

    [<Benchmark>]
    member this.PatternConsume() = pattern.Consume(this.Source)

    [<Benchmark(Baseline = true)>]
    member this.ParsecRun() = run parsec this.Source